[GauhVish] iP

GauhVish opened the pull request:
# AquaDem
> " The mind is not a vessel to be filled but a fire to be kindled." - Plutarch ([Source](https://peacefulsoul.co/the-mind/quotes-about-the-mind/))
Dont waste your time with reminders, let AquaDem do it for you.

- Text-based
- No programming knoweldge needed
- ~~Fast~~ INSTANT

Just follow these simple steps:
1. download it from [here](https://github.com/GauhVish/ip/releases)
2. open it up
3. add your tasks
4. never look back again!

Best part ? Its also **FREE**

- [x] Managing Task
- [x] Managing Deadlines
- [ ] Auto Reminder (coming soon)


If you are a Java programmer, you can use it to practice Java too. Here's the ```main ``` method:
``` ruby
    public static void main(String[] args) {
        AquaDem chatbot = new AquaDem(new Storage());
        chatbot.intro();
        chatbot.running();

    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Renaming main class and finishing Level-0

Commit:
Added some abstraction and implemented Level-1

Commit:
Level 2 Work

Commit:
Level-3

Commit:
Level-4 work

Commit:
A-TextUiTesting

Commit:
Level-5

Commit:
Level-5 rework

Commit:
Level-6

Commit:
Add save features and create separate Task class.

Commit:
Merge branch 'branch-Level-7'

Commit:
Add Date Support for Task

Commit:
Merge branch 'branch-Level-8'

Commit:
Separate functions into different classes

Commit:
Fix storage and delete bug

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Add and Fix package Aquadem

Commit:
Merge branch 'branch-A-Packages'

Commit:
Merge branch 'add-gradle-support'

Commit:
Add gradle support

Commit:
Add Junit support and test

Commit:
Merge branch 'branch-A-JUnit'

Commit:
Added javadocs comments

Commit:
Add CodingStyle edits

Commit:
Add find command

Commit:
Configure build file to produce .jar

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Aquadem/Parser.java
#	src/main/java/Aquadem/Storage.java
#	src/main/java/Aquadem/Task.java

Commit:
Merge branch 'branch-A-Level-9'

# Conflicts:
#	src/main/java/Aquadem/Execution.java
#	src/main/java/Aquadem/Parser.java

Bryce-3D commented on src/main/java/Aquadem/Event.java:
```diff
@@ -0,0 +1,35 @@
+ */
+public class Event extends Task implements Serializable {
+
+    protected String t1;
+
+    protected String t2;
```
Hi! I think that these variable names might not be descriptive enough based on [Section 4.6d of the notes](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/topics.html#w4-6-code-quality-naming).

GauhVish replied:
Hi i agree : ) Thank you!

Bryce-3D commented on src/main/java/Aquadem/Execution.java:
```diff
@@ -0,0 +1,123 @@
+     * @param date
+     * @return LocalDateTime corresponding to the date
+     */
+    public static LocalDateTime dateConvert(String date) {
```
Hi! I think that the function name `convertDate()` might be more appropriate here considering the examples given in [4.6d](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/topics.html#w4-6-code-quality-naming)

GauhVish replied:
Hi i agree : ) Thank you!

Bryce-3D commented on src/main/java/Aquadem/Parser.java:
```diff
@@ -0,0 +1,168 @@
+     * @param detail
+     * @throws DetailException
+     */
+    public static void detailCheck(String detail) throws DetailException {
```
Hi! I think that the function name `checkDetail` might be better here, since in English it would be read as "This function checks details," similar to the examples in [4.6d](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/topics.html#w4-6-code-quality-naming)

Bryce-3D replied:
I also have similar thoughts on most of the function names within this file

GauhVish replied:
Hi i agree : ) Thank you!

Bryce-3D reviewed (commented):
Hi! Overall, the naming here looks good to me.

Hope this review helps!

niha81 commented on src/main/java/Aquadem/Execution.java:
```diff
@@ -0,0 +1,123 @@
+     */
+    public static void execute(Pair encodedCommand, TaskList tasks, Storage storage){
+        int command = encodedCommand.getHead();
+        String[] parsedDetail = encodedCommand.getContents();
```
Would it be better for this string array need to be named parsedDetails?

niha81 commented on src/main/java/Aquadem/Ui.java:
```diff
@@ -0,0 +1,152 @@
+ * A class that encapsulates the User Interactions and visual output with the chatbot
+ */
+public class Ui {
+    protected final static String bar = "____________________________________________________________";
```
should the constant be named BAR?

niha81 reviewed (commented):
Overall it looks good to me! Perhaps you could put your files into more specific packages for easier navigation :)

Commit:
Add javafx gui and support

Commit:
Merge branch 'branch-A-Level-10'

Commit:
Add assertions for internal process verification

Assertions are needed to ensure that the code is internally consistent, It is not used for preexisting
exception handling, purely to ensure that internal processes are executed as expected.

Commit:
Add support for DoAfter tasks

DoAfter tasks are similar to Todo tasks except they are done after a date.
Add support for them now and perhaps later develop them with reminders

Commit:
Adjust javadocs for Event class

GauhVish committed:
Merge pull request #1 from GauhVish/branch-A-Assertions

Add assertions for internal process verification

Commit:
Merge branch 'master' into branch-A-CodeQuality

Commit:
Merge branch 'master' into branch-B-DoAfterTasks

GauhVish committed:
Merge pull request #2 from GauhVish/branch-B-DoAfterTasks

Branch b do after tasks

Commit:
Merge branch 'master' into branch-A-CodeQuality

GauhVish committed:
Merge pull request #3 from GauhVish/branch-A-CodeQuality

Branch a code quality

Commit:
Repair code quality and Add UG  Add a user guide and fix the code quality errors

Commit:
Finalise user guide and image

Commit:
Add Ui.png

Commit:
Update gradle file

Update the relevant build.gradle file

Commit:
Update storage class

The storage file needs to be cross-system compatible.

Commit:
Update readme file

The readme file needs to fix its image.

Commit:
Update javafx related files

Update some javafx files with new paths.

Commit:
Update userimage and readme

Updated the userimage, UI and readme files.
